commit c82c326f2cb63718491723b7aee09350a4bb90a5
Author: Vishal Verma <vishal.l.verma@intel.com>
Date:   Fri May 19 11:39:10 2017 +0200

    acpi, nfit: Fix the memory error check in nfit_handle_mce()
    
    commit fc08a4703a418a398bbb575ac311d36d110ac786 upstream.
    
    The check for an MCE being a memory error in the NFIT mce handler was
    bogus. Use the new mce_is_memory_error() helper to detect the error
    properly.
    
    Reported-by: Tony Luck <tony.luck@intel.com>
    Signed-off-by: Vishal Verma <vishal.l.verma@intel.com>
    Signed-off-by: Borislav Petkov <bp@suse.de>
    Link: http://lkml.kernel.org/r/20170519093915.15413-3-bp@alien8.de
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/acpi/nfit/mce.c b/drivers/acpi/nfit/mce.c
index e5ce81c..e25787a 100644
--- a/drivers/acpi/nfit/mce.c
+++ b/drivers/acpi/nfit/mce.c
@@ -26,7 +26,7 @@ static int nfit_handle_mce(struct notifier_block *nb, unsigned long val,
 	struct nfit_spa *nfit_spa;
 
 	/* We only care about memory errors */
-	if (!(mce->status & MCACOD))
+	if (!mce_is_memory_error(mce))
 		return NOTIFY_DONE;
 
 	/*
